### PR TITLE
feat: add confirmation modal component

### PR DESCRIPTION
**Pull request by frosso**
Helps with https://github.com/Automattic/woocommerce-payments/issues/1564

#### Changes proposed in this Pull Request

It looks like we're starting to use a few of these modals within our codebase.
I'd like to add a "confirmation modal" component, which takes care of the layout of the modals we're using in the codebase.

These are all the existing modals:
- https://d.pr/i/YvnEJu
- https://d.pr/i/d5D7Kn

And we're adding also this:
- https://d.pr/i/5l2or7 - which I'm doing for the confirmation dialog task: https://github.com/Automattic/woocommerce-payments/issues/1564

As you can see, they all have a:
- title
- body
- separator
- CTAs

The separator and the CTAs all need to be styled each time. I thought about creating a separate component where the styles can be defined once.
I thought that for the confirmation dialog task ( https://github.com/Automattic/woocommerce-payments/issues/1564 ) I would DRY up the code a bit.

#### Testing instructions

- Ensure you have the latest version of the dev tools: `cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools && git checkout trunk && git pull`
- Ensure you have the "grouped settings" flag enabled in WCPay Dev Tools
- Ensure you have the "Add UPE additional payment methods" flag enabled in WCPay Dev Tools
- Go to your WCPay Home: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout
- Clicking on "add payment method" or the "delete" button on each payment method displays the modal
- No change in functionality compared to the current `develop` branch

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/settings/payment-methods-selector/index.js [hunk 1]
```diff
@@ -103,19 +118,7 @@ const PaymentMethodsSelector = ( { className } ) => {
 							)
 						) }
 					</PaymentMethodCheckboxes>
-					<HorizontalRule className="woocommerce-payments__payment-method-selector__separator" />
-					<div className="woocommerce-payments__payment-method-selector__footer">
```
Comment (frosso) [inline client/settings/payment-methods-selector/index.js:107]: This (and its styles) used to be repeated

File: client/components/payment-methods-list/delete-button.js [hunk 2]
```diff
@@ -98,20 +99,7 @@ const DeleteButton = ( { id, label, Icon, onClick, className } ) => {
 							},
 						} ) }
 					</p>
-					<HorizontalRule className="payment-method-delete-modal__separator" />
-					<div className="payment-method-delete-modal__footer">
```
Comment (frosso) [inline client/components/payment-methods-list/delete-button.js:102]: This (and its styles) used to be repeated

File: client/components/payment-methods-list/delete-button.js [hunk 3]
```diff
@@ -3,18 +3,16 @@
...
 import interpolateComponents from 'interpolate-components';
-import Gridicon from 'gridicons';
-import { HorizontalRule } from '@wordpress/primitives';
 import { useCallback, useState } from '@wordpress/element';
-import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-import './delete-button.scss';
```
Comment (frosso) [inline client/components/payment-methods-list/delete-button.js:17]: No longer need separate styles

File: client/settings/payment-methods-selector/index.js [hunk 4]
```diff
@@ -16,7 +15,7 @@ import {
 } from 'data';
 import PaymentMethodCheckboxes from '../../components/payment-methods-checkboxes';
 import PaymentMethodCheckbox from '../../components/payment-methods-checkboxes/payment-method-checkbox';
-import './style.scss';
```
Comment (frosso) [inline client/settings/payment-methods-selector/index.js:19]: No longer need separate styles

**Review by Ferdev — APPROVED** [review]
LGTM!!

**Pull request merged**